mh/session.py

```diff
@@ -190,7 +190,8 @@ def layer_down(self, layer_id):
     def layer_create(self, layer_id, settings, optional_fields):
         if self.layer == 1:
             city = self.create_city(layer_id, settings, optional_fields)
-            city.leader = self
+            with city.lock():
```

Comment:
I don't think that you need to lock the city that you just created. 

Now that I think about it, why not move the assignment of the leader to the `create_city` method?

Reply:
All writes should have locks on them. However I think you're right, it would be better to have the city leader be assigned inside the `create_city` method. After all, no cities can exist without leaders.

Reply:
Changed this and pushed.